annotation-utils/src/main/java/edu/isi/vista/annotationutils/ParseEventLogs.kt

```diff
@@ -100,24 +100,41 @@ class ParseEventLogs {
                     logger.info { "Skipping $projectName"}
                 }
             }
-            // If there is a directory of original log files, include that data
+            // If there is a directory of original log files, include that data.
+            // First, create a mapping of eventType/usernames to project info
+            // in order to select the right project to modify.
+            val eventAndUserToProjectInfo = allProjectInfo.map {
+                Pair(it.eventType, it.username) to it
+            }.toMap().toMutableMap()
```

Comment:
Can a `MutableMap` only be constructred from  `Map`?

Reply:
Yeah. The `.map()` function here applies a lambda function and creates a `List` of `Pair` objects which can then be converted to a `Map` -- `toMutableMap()` can only be performed on a `Map` object.